### PR TITLE
Test 20.05.23 2

### PR DESCRIPTION
**Pull request by geopin90**

### CODE REVIEW

File: tests/check_backdated_set_cost.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+use service::utils::json::JsonParams;
+use service::{Context, Services};
+use store::balance::{BalanceForGoods, Cost};
+use store::batch::Batch;
+use store::elements::{dt, Goods, Mode, Qty, Store};
+use store::operations::{InternalOperation, OpMutation};
+use store::process_records::process_record;
+use store::GetWarehouse;
+
+const WID: &str = "yjmgJUmDo_kn9uxVi8s9Mj9mgGRJISxRt63wT46NyTQ";
```
Comment (shabanovd) [inline tests/check_backdated_set_cost.rs:25]: common one must be used

File: tests/check_backdated_set_cost.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+  log::debug!("transfer 26.01 s1 > s2 25");
+  transfer(&app, "2023-01-26", s1, s2, g1, 25.into());
+
+  log::debug!("receive 20.01 s1 25");
+  let r1 = receive(&app, "2023-01-20", s1, g1, 25.into(), "25".try_into().unwrap());
+  let r1_batch = Batch { id: r1, date: dt("2023-01-20").unwrap() };
+
+  // s1 b0 0 0
+  // s2 b0 25 25
```
Comment (shabanovd) [inline tests/check_backdated_set_cost.rs:56]: this is incorrect

File: tests/check_backdated_set_cost.rs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  // s2 r1 25 25
+
+  let balances = app.warehouse().database.get_balance_for_all(Utc::now()).unwrap();
+  log::debug!("balances: {balances:#?}");
+
+  assert_eq!(balances.len(), 1);
+
+  let s1_bs = balances.get(&s1);
+  assert_eq!(s1_bs, None);
```
Comment (shabanovd) [inline tests/check_backdated_set_cost.rs:65]: no needs to do such check

File: tests/check_backdated_set_cost.rs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+use store::batch::Batch;
+use store::elements::{dt, Goods, Mode, Qty, Store};
+use store::operations::{InternalOperation, OpMutation};
+use store::process_records::process_record;
+use store::GetWarehouse;
+
+const WID: &str = "yjmgJUmDo_kn9uxVi8s9Mj9mgGRJISxRt63wT46NyTQ";
+
+#[actix_web::test]
+async fn check_backdated_set_cost() {
```
Comment (shabanovd) [inline tests/check_backdated_set_cost.rs:28]: what kind of new checks here vs `check_zero_batch_deleted`?

Comment (geopin90) [reply tests/check_backdated_set_cost.rs:28]: same

**Pull request closed without merging**